recipes/libftdi/1.x/patches/0001-cmake-targets.patch

```diff
@@ -0,0 +1,787 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 5aecafc..aa69995 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,36 +1,54 @@
+-# Project
+ project(libftdi1 C)
++
+ set(MAJOR_VERSION 1)
+ set(MINOR_VERSION 5)
+ set(PACKAGE libftdi1)
+ set(VERSION_STRING ${MAJOR_VERSION}.${MINOR_VERSION})
+ set(VERSION ${VERSION_STRING})
+-set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
++list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+ 
+-# CMake
+ if("${CMAKE_BUILD_TYPE}" STREQUAL "")
+    set(CMAKE_BUILD_TYPE     RelWithDebInfo)
+ endif("${CMAKE_BUILD_TYPE}" STREQUAL "")
+ set(CMAKE_COLOR_MAKEFILE ON)
+-cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
++
++cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
++if(POLICY CMP0057)
++    cmake_policy(SET CMP0057 NEW)
++endif()
++if(POLICY CMP0011)
++    cmake_policy(SET CMP0011 NEW)
++endif()
++if(POLICY CMP0099)
++    cmake_policy(SET CMP0099 NEW)
++endif()
+ 
+ add_definitions(-Wall)
+ 
+ include(CMakeOptions.txt)
+ 
++set(as_subproject LibFTDI)
++macro(find_package)
```

Comment:
Overriding the cmake function? 😕 Why is this needed?

Reply:
This allows the library to be consumed by examples and applications in the build tree as if it were an external project, and is considered CMake best practice. See [Effective CMake by Daniel Pfeiffer (CPPNow 2017) Slide 31](https://github.com/boostcon/cppnow_presentations_2017/blob/master/05-19-2017_friday/effective_cmake__daniel_pfeifer__cppnow_05-19-2017.pdf)